### DISCUSSION TITLE
question about migrating from opentracing to opentelemetry

### DISCUSSION THREAD

**Discussion by operations-ivy** [jaegertracing/jaeger#2803] [Q&A]
Hello, we're in the process of replacing our existing jaeger agents and collectors with jaeger opentelemetry agents & collectors. During this migration we plan on using the same cassandra cluster for a backend.

My question is this, if we're using the same cassandra cluster, should we expect any issues if the jaeger collector AND the opentelemetry jaeger collector are working on the same machine (different ports)?

One thing we suspect might be an issue is new opentelemetry traces using the same traceid that is already in cassandra from the older collector. Will this happen? If it will happen can it be avoided?

Thanks!

Patrick

**Comment by yurishkuro** [comment]
I don't foresee any issues. Trace IDs are assigned by instrumentation, not collectors. I assume you're not sending traces simultaneously to both types of collectors.

Which `opentelemetry jaeger collector` are you using though? We don't have them as official releases.

**Reply by operations-ivy** [reply]
I thought I saw a link to the github releases page that had those binaries but after looking again I don't see them! I do see a mention of the opentelemetry jaeger binaries being made available in the future, is there an estimate of when that will be?

**Reply by yurishkuro** [reply]
We had an experimental set of binaries, but the way they were built did not pan out strategically, so there is a different Jaeger V2 effort going (in https://github.com/jaegertracing/jaeger-otelcol). We don't have an estimate on completion.

Having said that, if you're interested in changing instrumentation from OpenTracing to OpenTelemetry, then none of this really matters, you can run the official OTEL collector as agent and the classic Jaeger backend, they work together fine. The point of Jaeger on OpenTelemetry work is more about internal infrastructure, being able to produce smaller binary for the agent than official OTEL collector, and being able to receive OTLP protocol directly.

**Reply by bboreham** [reply]
I did find it useful to tell people to use the all-in-one docker image (`docker.io/jaegertracing/opentelemetry-all-in-one`) as a step to get started, since it reduces the set of things that can go wrong. (I did know it was marked as experimental).

Is there another all-in-one option where people can receive OTLP traces and view them?

**Reply by yurishkuro** [reply]
Not at this time, unfortunately.  We found OTEL collector code to be too volatile to serve as a base of jaeger v2 development. We will revisit it once collector is stable.
